ffi/derive/src/attr_parse/getset.rs

```diff
@@ -0,0 +1,830 @@
+//! This module provides parsing of custom attributes from the [`getset`](https://docs.rs/getset/latest/getset/) crate
+
+use std::{collections::hash_map::Entry, fmt::Display, str::FromStr};
+
+use proc_macro2::Span;
+use rustc_hash::{FxHashMap, FxHashSet};
+use syn2::{parse::ParseStream, punctuated::Punctuated, Attribute, Token};
+
+use crate::attr_parse::derive::{Derive, DeriveAttr};
+
+/// Type of accessor method derived for a structure
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum GetSetDerive {
+    Setters,
+    Getters,
+    MutGetters,
+    CopyGetters,
+}
+
+impl GetSetDerive {
+    pub fn try_from_path(path: &syn2::Path) -> Option<Self> {
+        // try to be smart and handle two cases:
+        // - bare attribute name (like `Getters`, when it's imported)
+        // - fully qualified path (like `getset::Getters`, when it's not imported)
+        let ident = if let Some(i) = path.get_ident() {
+            i.clone()
+        } else {
+            let mut segments = path.segments.iter();
+            if segments.len() == 2
+                && segments.next().unwrap().ident.to_string().as_str() == "getset"
+            {
+                segments.next().unwrap().ident.clone()
+            } else {
+                return None;
+            }
+        };
+
+        match ident.to_string().as_str() {
+            "Setters" => Some(Self::Setters),
+            "Getters" => Some(Self::Getters),
+            "MutGetters" => Some(Self::MutGetters),
+            "CopyGetters" => Some(Self::CopyGetters),
+            _ => None,
+        }
+    }
+
+    pub fn get_mode(self) -> GetSetGenMode {
+        match self {
+            Self::Setters => GetSetGenMode::Set,
+            Self::Getters => GetSetGenMode::Get,
+            Self::MutGetters => GetSetGenMode::GetMut,
+            Self::CopyGetters => GetSetGenMode::GetCopy,
+        }
+    }
+}
+
+#[derive(Default, Debug, Eq, PartialEq, Clone)]
+pub struct GetSetOptions {
+    pub visibility: Option<syn2::Visibility>,
+    pub with_prefix: bool,
+}
+
+struct SpannedGetSetOptions {
+    span: Span,
+    options: GetSetOptions,
+}
+
+impl syn2::parse::Parse for SpannedGetSetOptions {
+    fn parse(input: ParseStream) -> syn2::Result<Self> {
+        let mut result = GetSetOptions::default();
+        // an accumulator for syn errors?
+        // this is getting out of hand...
+        // we need an accumulator to rule them all!
+        let mut errors = Vec::new();
```

Comment:
Shouldn't you use a `drop_bomb` here?

Reply:
Yeah, the usual accumulator facilities would be nice. But, as it's a one-off thing (at least for now), it would be fine I think. There are no early returns in this function, so the errors will always be handled in this case